maap/maap.py

```diff
@@ -195,6 +201,64 @@ def getJobStatus(self, jobid):
         )
         return response
 
+    def executeQuery(self, src, query={}, results=True, timeout=180, wait_interval=.5):
+        """
+        Helper to execute query and poll results URL until results are returned
+        or timeout is reached.
+
+        src -- a dict-like object stipulating which dataset is to be queried.
+            Object must contain either a 'Granule' or 'Collection' key.
+            Collection-related value must contain 'ShortName' and 'VersionId'
+            entries. Granule-related value must contain a 'Collection' entry,
```

Comment:
Does a granule make sense? I recall having the thought that it really only makes sense to query the collection (and then you can do queries on the collection that could get at data that originated from a granule file like for site name or project)

I'm thinking when developing the feature of "querying a collection via API" we discard the concept of granules.